producers/branch_data/branch_producer.py

```diff
@@ -0,0 +1,131 @@
+""" Branch data producer that creates dummy branch data """
+import sys
+import os
+import json
+from dotenv import load_dotenv
+from faker import Faker
+import requests
+
+
+class BranchProducer:
+    """
+    BranchProducer that will create dummy data for the branch microservice
+    takes various parameters
+    -d to delete all records
+    -l to list all branches
+    -n to denote the number of records to insert
+    """
+
+    def __init__(self, n_records=0):
+        self.n_records = int(n_records)
+        self.records_produced = []
+
+        load_dotenv()
+        self.base_url = os.getenv("BASE_URL")
+        self.insert_url = os.getenv("INSERT_URL")
+        self.get_url = os.getenv("GET_URL")
+        self.delete_url = os.getenv("DELETE_URL")
+
+    def __repr__(self):
+        return f"Branch Producer to generate {self.n_records} records"
+
+    def verify_records(self, total_start):
+        """ verify that the total items in the record match the amount to insert or delete """
+        total_end = self.get_all_branches()["totalItems"]
+        if total_start + self.n_records == total_end:
+            return True
+        return False
+
+    def branch_generator(self):
+        """ with faker library will generate n records """
+        for _ in range(self.n_records):
+            fake = Faker()
+            branch_name = fake.last_name() + fake.last_name()
+            address = fake.address().split("\n")
+            address_line = address[0]
+            city = address[1].split(",")[0]
+            address_part_2 = address[1].split()
+            state = address_part_2[-2]
+            zipcode = address_part_2[-1]
+            branch_details = fake.paragraph(nb_sentences=2)
+
+            branch_dict = {
+                "branchName": branch_name,
+                "addressLine1": address_line,
```

Comment:
There's also an "addressLine2" field in the Branch Address model; I think it can be null, but I'm not sure if leaving the field out of the request will cause an error on the back end. Might be worth adding it in for variation as well. A typical line 2 would probably be something like "Suite ###"

Reply:
Yeah it submits fine. I just modeled it off what Alex had sent me. it appears to request fine. like if you fetch the data after inserting. it works alright. I'll go ahead and it add for completeness.

Reply:
alright added it. 25% chance to have that field otherwise it's N/A.